source/common/http/conn_manager_impl.cc

```diff
@@ -158,6 +158,8 @@ void ConnectionManagerImpl::initializeReadFilterCallbacks(Network::ReadFilterCal
       {stats_.named_.downstream_cx_rx_bytes_total_, stats_.named_.downstream_cx_rx_bytes_buffered_,
        stats_.named_.downstream_cx_tx_bytes_total_, stats_.named_.downstream_cx_tx_bytes_buffered_,
        nullptr, &stats_.named_.downstream_cx_delayed_close_timeout_});
+
+  filter_state_ = read_callbacks_->connection().streamInfo().filterState();
```

Comment:
I think this works for reading from connection filter state but not for writing. To make it work for writing, we would need to expose a non-const filter state and change the parent level to "connection". Right now connection filter state level is set to "request" (by mistake).

Reply:
I gave this a shot, but instead of exposing a mutable ref I changed the LazyAncestor to hold the shared ptr directly instead of a ref. Not sure if there's a good reason why the ref was used instead of the shared_ptr to begin with

Reply:
Not sure either. There isn't much value for lazy creation here since all state is concrete (connection, filter_chain, request).